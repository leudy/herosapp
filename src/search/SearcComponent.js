import React from "react";

export const SearcComponent = () => {
  return <div></div>;
};
