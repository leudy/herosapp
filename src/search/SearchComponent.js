import React, { useMemo } from "react";
import { useLocation } from "react-router-dom";
import queryString from "query-string";

import { useForm } from "../hooks/useForm";
import { HeroeCard } from "../heroes/HeroeCard";
import { getHeroeByFilter } from "../repositories/HeroesRepository";

export const SearchComponent = ({ history }) => {
  const location = useLocation();
  const StringQuery = queryString.parse(location.search);
  const { q = "" } = StringQuery;
  const initialState = {
    criteriaText: q,
  };
  const [form, handleTextChange] = useForm(initialState);
  const data = useMemo(() => getHeroeByFilter(q), [q]);
  const { criteriaText } = form;
  
  const handleSubmit = (f) => {
    f.preventDefault();
    const filter = `?q=${criteriaText}`;
    history.push(filter);
  };

  return (
    <div>
      <h1>Search Screem</h1>
      <hr />
      <div className="row">
        <div className="col-md-5">
          <h4> Search form </h4>
          <hr />
          <form onSubmit={handleSubmit}>
            <div className="form-group">
              <input
                value={criteriaText}
                onChange={handleTextChange}
                type="text"
                name="criteriaText"
                className="form-control"
              />
            </div>
            <button type="submit" className="btn btn-outline-primary btn-block">
              Find
            </button>
          </form>
        </div>
        <div className="col-md-7">
          <h4>Result:</h4>
          <hr />
          {q === "" && (
            <div className="alert alert-info">
              <p>Search a heroe</p>
            </div>
          )}

          {q !== "" && data.length === 0 && (
            <div className="alert alert-danger">
              <p>
              there is no a heroe with {q} </p>
            </div>
          )}

          {data.map((h) => (
            <div className="animate__animated animate__bounceInUp" key={h.id}>
              <HeroeCard key={h.id} {...h}></HeroeCard>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};
