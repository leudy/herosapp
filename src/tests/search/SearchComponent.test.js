import React from 'react'
import { MemoryRouter, Route } from 'react-router-dom'
import { SearchComponent } from '../../search/SearchComponent'
import Adapter from "enzyme-adapter-react-16"
import { mount,  configure} from "enzyme";   
import '@testing-library/jest-dom'
configure({adapter: new Adapter()});

describe('Probando el <Search />', () => {
  
  const wrapper = mount(
         <MemoryRouter initialEntries={['/search']}>
             <Route path='/search' component={SearchComponent}>  
             </Route>
         </MemoryRouter>
  )


     test('Probar el display por default', () => {
        expect(wrapper).toMatchSnapshot();
     })
     test('Probando que muestro texto de buscar', () => {
       expect(wrapper.find('.alert-info').text().trim()).toBe('Search a heroe')
     })

     test('Probar que la busqueda tenga el mismo valor que la url', () => {
            const _queryWrapper = mount(
                            <MemoryRouter initialEntries={['/search?q=superman']}>
                                    <Route path='/search' component={SearchComponent}>  
                                    </Route>
                            </MemoryRouter>
                    )
            expect(
                _queryWrapper.find('input').prop('value')
            ).toBe('superman');

     })

    
    
     test('should be a text error if url is wrong like as232212', () => {
        const _queryWrapper = mount(
            <MemoryRouter initialEntries={['/search?q=superman123']}>
                    <Route path='/search' component={SearchComponent}>  
                    </Route>
            </MemoryRouter>
    );
     console.log(_queryWrapper.find('.alert-danger').text().trim(),'aqui debe decir algo')
     expect(_queryWrapper.find('.alert-danger').text().trim()).toBe('there is no a heroe with superman123')
     //console.log(_queryWrapper.html())
     expect(_queryWrapper).toMatchSnapshot();
     
        })

     
     test('Debe de llmar el push del history ', () => {
    
     // config wrapper and hisortiry
        const history = {
        push:jest.fn()
    }  
            const wrapper = mount(
                <MemoryRouter initialEntries={['/search?q=batman123']}>
            <Route path="/search"
            component={()=> <SearchComponent history={history} />}
            >
            </Route>
                </MemoryRouter>
            )
    

    wrapper.find('input').simulate('change',{
    target:{
        name:'searchText',
        value:'batman'
    }
    });
    // make test
    wrapper.find('form').prop('onSubmit')({

         preventDefault(){}
    })
   //      console.log(wrapper.html())
     expect(history.push).toHaveBeenCalledWith('?q=batman123');





     })
     
     
     





})
