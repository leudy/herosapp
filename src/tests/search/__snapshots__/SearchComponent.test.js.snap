// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el <Search /> Probar el display por default 1`] = `ReactWrapper {}`;

exports[`Probando el <Search /> should be a text error if url is wrong like as232212 1`] = `ReactWrapper {}`;
